tests/test.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/sh
+
+set -x
+DEVFILES_DIR="$(pwd)/stacks/"
+FAILED_TESTS=""
+
+getURLs() {
+    urls=$($ODO_PATH url list | awk '{ print $3 }' | tail -n +3 | tr '\n' ' ')
+    echo "$urls"
+}
+
+# periodicaly check url till it returns expected HTTP status
+# exit after 10 tries
+waitForHTTPStatus() {
+   url=$1
+   statusCode=$2
+
+    for i in $(seq 1 10); do
+        echo "try: $i"
+        content=$(curl -i "$url")
+        echo "Checking if $url is returning HTTP $statusCode"
+        echo "$content" | grep -q -E "HTTP/[0-9.]+ $statusCode"
+        retVal=$?
+        if [ $retVal -ne 0 ]; then
+            echo "ERROR not HTTP $statusCode"
+            echo "$content"
+        else
+            echo "OK HTTP $statusCode"
+            return 0
+        fi
+        sleep 10
+    done
+    return 1
+}
+
+# run test on devfile
+# parameters:
+# - name of a component and project 
+# - path to devfile.yaml
+test() {
+    devfileName=$1
+    devfilePath=$2
+
+    # remember if there was en error
+    error=false
+
+    tmpDir=$(mktemp -d)
+    cd "$tmpDir" || return 1
+
+    $ODO_PATH project create "$devfileName" || error=true
+    if $error; then
+        echo "ERROR project create failed"
```

Comment:
Should we add the cleanup step after we hit the error? Probably we can create a clean up function and call it either the test fails in the middle or the test succeeds in the end.

Reply:
I'd say no, it's in a temp folder that will be removed upon reboot/restart.